app/services/s3_handler.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require 'aws-sdk-s3'
+
+class S3Handler
+  INPUT_PREFIX = 'pdf/'
+  OUTPUT_PREFIX = 'result/COMPLIANT_'
+
+  def initialize(object_key)
+    @object_key = object_key
+    @s3 = Aws::S3::Resource.new(
+      region: ENV.fetch('AWS_REGION'),
+      credentials: Aws::Credentials.new(
+        ENV.fetch('AWS_ACCESS_KEY_ID'),
+        ENV.fetch('AWS_SECRET_ACCESS_KEY')
+      )
+    )
+    @bucket = @s3.bucket(ENV.fetch('S3_BUCKET_NAME'))
+  end
+
+  def upload_to_input(local_path)
+    key = "#{INPUT_PREFIX}#{@object_key}"
+    @bucket.object(key)
+      .upload_file(local_path)
+  end
+
+  def presigned_url_for_output(expires_in: 3600)
+    obj = find_file(prefix: OUTPUT_PREFIX)
+    return nil unless obj
+
+    obj.presigned_url(:get, expires_in: expires_in)
+  end
+
+  def delete_files
```

Comment:
How are we thinking that cleanup of old files will happen? The PDF remediation app itself doesn't remove the input file once it has generated a corresponding output file, correct?

Is there a good way for this app to know when a client has downloaded the output file (assuming that they only need to download it once)? Or do we think that we'll just leave the files around for a reasonable length of time (maybe after the original pre-signed URL has expired) and then clean them up on a schedule?